cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -535,6 +535,7 @@ struct BasicTest {
         large_utf8(),
         binary(),
         large_binary(),
+        boolean(),
```

Comment:
Remind me how this works? This will randomly generate a boolean column?

Reply:
`all_types` defines the union of all on-key/by-key/data types that is later filtered to obtains several lists. Each list is used to span a dimension in a sample space; that is, a sample is a combination, one type from each such list. This space is sampled up to 100 in a loop.

Reply:
No, it won't generate a random column, just a random type for the column.

Reply:
What value does it use to fill in boolean column?

Reply:
It ends up converting any non-zero numeric value in the test data to `true`. Since all relevant values are non-zero, this is not good coverage. I just committed a fix for this.